packages/subiquity_client/lib/src/server.dart

```diff
@@ -129,35 +61,5 @@ abstract class SubiquityServer {
     client.close();
   }
 
-  Future<void> stop() async {
-    try {
-      await pidFile().delete();
-    } on FileSystemException catch (_) {}
-    _serverProcess?.kill();
-    await _serverProcess?.exitCode;
-  }
-}
-
-class _SubiquityServerImpl extends SubiquityServer {
-  _SubiquityServerImpl() : super._();
-
-  // Normally, the server is already running in live mode and thus, only
-  // started in dry-run mode.
-  @override
-  bool _shouldStart(ServerMode mode) => mode == ServerMode.DRY_RUN;
-
-  @override
-  String get _pythonModule => 'subiquity.cmd.server';
-}
-
-// A server that runs in a WSL environment.
-class _WslSubiquityServerImpl extends SubiquityServer {
-  _WslSubiquityServerImpl() : super._();
-
-  // The server should be always started in WSL because there's no systemd.
-  @override
-  bool _shouldStart(ServerMode mode) => true;
-
-  @override
-  String get _pythonModule => 'system_setup.cmd.server';
+  Future<void> stop() async => await launcher?.stop();
```

Comment:
```suggestion
  Future<void> stop() async => launcher?.stop();
```
the await is not necessary inside stop, we can return the future and let the caller decide (one could e.g. await multiple futures in parallel with Future.wait([...]))

Reply:
Good point. Fixed.